trie/trie.go

```diff
@@ -48,6 +48,14 @@ type LeafCallback func(leaf []byte, parent common.Hash) error
 type Trie struct {
 	db   *Database
 	root node
+	// Keep a track of the number of leafs to commit. This counter is
+	// update for inserts and deletions of leafs, but does not accurately track the number
+	// of nodes
+	dirtyCount int
+	// Keep track of the number leafs which have been inserted since the last
+	// hashing operation. This number will not directly map to the number of
+	// actually unhashed nodes
+	unhashedCount int
```

Comment:
Maaaaybe rename to `unhashed`? :D I'm not a fan of explicitly stating `counter` unless it makes things extra super clear. But up to you really :P